src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,49 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+
+    private Injector() {
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (!interfaceClazz.isAnnotationPresent(Component.class)) {
```

Comment:
The check for `@Component` annotation assumes that the provided class is a concrete class. If `interfaceClazz` is an interface, this check will fail. Consider handling interfaces differently, possibly by mapping interfaces to their implementations.